client/multi-currency/blocks/currency-switcher.js

```diff
@@ -0,0 +1,318 @@
+			setAttributes( {
+				borderColor:
+					newBorderColor === undefined ? '#000000' : newBorderColor,
+			} );
+		};
+
+		const onChangeBackgroundColor = ( newBackgroundColor ) => {
+			setAttributes( {
+				backgroundColor:
+					newBackgroundColor === undefined
+						? 'transparent'
+						: newBackgroundColor,
+			} );
+		};
+
+		return (
+			<div { ...blockProps }>
+				<InspectorControls key="setting">
+					<PanelBody
+						title={ __(
+							'Multi-Currency Settings',
+							'woocommerce-payments'
+						) }
+					>
+						<a
+							href={
+								'/wp-admin/admin.php?page=wc-settings&tab=wcpay_multi_currency'
+							}
+							target="_blank"
+							rel="noreferrer"
+						>
+							Adjust and edit your Multi-Currency settings
+						</a>
+					</PanelBody>
+					<PanelBody title={ __( 'Layout', 'woocommerce-payments' ) }>
+						<CheckboxControl
+							label={ __(
+								'Display flags',
+								'woocommerce-payments'
+							) }
+							checked={ flag }
+							onChange={ onChangeFlag }
+						/>
+
+						<CheckboxControl
+							label={ __(
+								'Display currency symbols',
+								'woocommerce-payments'
+							) }
+							checked={ symbol }
+							onChange={ onChangeSymbol }
+						/>
+
+						<CheckboxControl
+							label={ __( 'Border', 'woocommerce-payments' ) }
+							checked={ border }
+							onChange={ onChangeBorder }
+						/>
+
+						<RangeControl
+							label={ __(
+								'Border radius',
+								'woocommerce-payments'
+							) }
+							value={ borderRadius }
+							onChange={ onChangeBorderRadius }
+							min={ 1 }
+							max={ 20 }
+						/>
+					</PanelBody>
+					<PanelBody
+						title={ __( 'Typography', 'woocommerce-payments' ) }
+					>
+						<RangeControl
+							label={ __( 'Size', 'woocommerce-payments' ) }
+							value={ fontSize }
+							onChange={ onChangeFontSize }
+							min={ 6 }
+							max={ 48 }
+						/>
+						<RangeControl
+							label={ __(
+								'Line height',
+								'woocommerce-payments'
+							) }
+							value={ fontLineHeight }
+							onChange={ onChangeFontLineHeight }
+							min={ 1 }
+							max={ 3 }
+							step={ 0.1 }
+						/>
+					</PanelBody>
+					<PanelBody
+						title={ __( 'Color settings', 'woocommerce-payments' ) }
+					>
+						<ColorPaletteControl
+							onChange={ onChangeFontColor }
+							value={ fontColor }
+							label={ __( 'Text', 'woocommerce-payments' ) }
+						/>
+
+						<ColorPaletteControl
+							onChange={ onChangeBackgroundColor }
+							value={ backgroundColor }
+							label={ __( 'Background', 'woocommerce-payments' ) }
+						/>
+
+						<ColorPaletteControl
+							onChange={ onChangeBorderColor }
+							value={ borderColor }
+							label={ __( 'Border', 'woocommerce-payments' ) }
+						/>
+					</PanelBody>
+				</InspectorControls>
+
+				<div className="currency-switcher-holder" style={ styles.div }>
+					<select name="currency" style={ styles.select }>
```

Comment:
I think, if we are using placeholders (meaning we couldn't fetch the enabled rates) instead, the select should be disabled to prevent user selecting a currency, and display a notice on the bottom (this might be optional). What will happen when the merchant didn't enable those currencies? The currency won't change AFAIK. Right?

Reply:
> I think, if we are using placeholders (meaning we couldn't fetch the enabled rates) instead, the select should be disabled to prevent user selecting a currency, and display a notice on the bottom (this might be optional). What will happen when the merchant didn't enable those currencies? The currency won't change AFAIK. Right?

So the placeholders were added by @dmallory42 when he was working on it because, one of the errors that he had, is that he was not able to retrieve the enabled currencies. I've mantained it there because without them, when loading the block it would appear empty. However, the currencies that are displayed there, if they are placeholders or not, are not affecting anything. They are just displayed to see how the currency switcher will look like, but it's not doing anything in terms of selecting the currency. It's just visual.

Reply:
I think it'll be better showing an empty select with a placeholder option saying "Loading..." instead showing currencies that might not be enabled, until the real ones are loaded. What do you think?